### PR TITLE
Rename all hash methods to match naming in John the Ripper.

### PR DESCRIPTION
**Pull request by besser82**
As the title says.

Any other things left to rename?  What about README.md and/or other docs besides `crypt.5`?

### CODE REVIEW

**Comment by besser82** [comment]
Shall we adopt {upper,lower}case like in JtR?  That would be for `sunmd5` -> `SunMD5` and `nt` -> `NT`.

**Comment by besser82** [comment]
@solardiz  Do you have any additional thoughts about this change?

File: .travis.yml [hunk 1]
```diff
@@ -216,47 +216,47 @@ matrix:
...
+        - CONF="--disable-obsolete-api --enable-hashes=bsdicrypt"
         - FCVER="rawhide"
-    - name: "Fedora Rawhide, GCC, des_xbsd only"
+    - name: "Fedora Rawhide, GCC, descrypt only"
       compiler: gcc
       os: linux
       services: docker
       env:
-        - CONF="--disable-obsolete-api --enable-hashes=des_xbsd"
+        - CONF="--disable-obsolete-api --enable-hashes=descrypt"
```
Comment (solardiz) [inline .travis.yml]: Perhaps this should be `bsdicrypt` (in two places above).

File: .travis.yml [hunk 2]
```diff
@@ -216,47 +216,47 @@ matrix:
...
         - CONF="--disable-obsolete-api --enable-hashes=bcrypt"
         - FCVER="rawhide"
-    - name: "Fedora Rawhide, GCC, des only"
+    - name: "Fedora Rawhide, GCC, bigcrypt only"
       compiler: gcc
       os: linux
       services: docker
       env:
-        - CONF="--disable-obsolete-api --enable-hashes=des"
+        - CONF="--disable-obsolete-api --enable-hashes=bigcrypt"
```
[the remark this answers is not in the document]
Comment (zackw) [reply .travis.yml]: Oh, nice catch, I bet this is why the coverage changed.

**Comment by solardiz** [comment]
> Shall we adopt {upper,lower}case like in JtR? That would be for `sunmd5` -> `SunMD5` and `nt` -> `NT`.

JtR has case-insensitive matching of these names, so accepts either/mixed case, but for hashes that were previously written in a certain way we try to write them the same. The capitalization of SunMD5 is by its author as seen e.g. in this blog post from 2005: https://dropsafe.crypticide.com/article/1389 although the Solaris source code appears to use all-lowercase as well. I don't feel about this strongly.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (besser82) [reply .travis.yml]: This is correct this way, as is changed with alphabetical reordering of the single methods.

See: https://github.com/besser82/libxcrypt/pull/39/files#diff-354f30a63fb0907d4ad57269548329e3R219

**Comment by solardiz** [comment]
Thanks @besser82! BTW, Fedora's libxcrypt.spec https://src.fedoraproject.org/rpms/libxcrypt/blob/master/f/libxcrypt.spec still says "SHA-2-512, SHA-2-256, SHA-1, MD5 (two variants), DES (three variants)" and probably needs a similar update.

**Pull request merged**